Assets/Scripts/UI/LevelSelectionHandle.cs

```diff
@@ -42,6 +42,8 @@ private void HideButtonsFromList()
 
     public void OpenLevel(int indexFromOne)
     {
+        // resetting deaths counter
+        DeathsCounter.instance.ResetCounters();
```

Comment:
Would be a lot better to use DontDestroyOnLoad, instead of singletonlike, could implement everything in one place